src/equations/ideal_glm_mhd_multiion.jl

```diff
@@ -263,4 +271,204 @@ end
+
+        h51 = h15
+        h52 = h25
+        h53 = h35
+        h54 = h45
+        h55 = ((p_star^2 / (gammas[k] - 1) + E_bar * E_bar) / rho_ln
+               +
+               vel_avg_norm * p_mean)
+        d5 = -0.5f0 * λ *
+             (h51 * (w1_rr - w1_ll) +
+              h52 * (w2_rr - w2_ll) +
+              h53 * (w3_rr - w3_ll) +
+              h54 * (w4_rr - w4_ll) +
+              h55 * (w5_rr - w5_ll))
+
+        beta_plus_ll += beta_ll
+        beta_plus_rr += beta_rr
+
+        set_component!(dissipation, k, d1, d2, d3, d4, d5, equations)
+    end
+
+    # Compute the dissipation related to the magnetic and divergence-cleaning fields
+    ################################################################################
+
+    h_B_psi = 1 / (beta_plus_ll + beta_plus_rr)
+
+    # Dissipation for the magnetic field components due to the diagonal entries of the 
+    # dissipation matrix ``H``. These are the dissipation values that depend on the diagonal
+    # entries of the entropy Jacobian that are marked in cyan in Figure 1 of the reference given above.
+    dissipation[1] = -0.5f0 * λ * h_B_psi * (w_rr[1] - w_ll[1])
+    dissipation[2] = -0.5f0 * λ * h_B_psi * (w_rr[2] - w_ll[2])
+    dissipation[3] = -0.5f0 * λ * h_B_psi * (w_rr[3] - w_ll[3])
+
+    # Dissipation for the divergence-cleaning field due to the diagonal entry of the 
+    # dissipation matrix ``H``. This dissipation value depends on the single diagonal
+    # entry of the entropy Jacobian that is marked in red in Figure 1 of the reference given above.
+    dissipation[end] = -0.5f0 * λ * h_B_psi * (w_rr[end] - w_ll[end])
+
+    # Dissipation due to the off-diagonal blocks (``B_{off}``) of the dissipation matrix ``H`` and to the entries 
+    # of the block ``A`` that depend on the magnetic field ``B`` and the divergence cleaning field ``psi``. 
+    # See equations (80)-(82) of the given reference.
+    for k in eachcomponent(equations)
+        _, _, _, _, w5_ll = get_component(k, w_ll, equations)
+        _, _, _, _, w5_rr = get_component(k, w_rr, equations)
+
+        # Dissipation for the magnetic field components and divergence cleaning field due to the off-diagonal 
+        # entries of the dissipation matrix ``H`` (block ``B^T`` in equation (80) and Figure 1 of the reference
+        # given above).
+        dissipation[1] -= 0.5f0 * λ * h_B_psi * B1_avg * (w5_rr - w5_ll)
+        dissipation[2] -= 0.5f0 * λ * h_B_psi * B2_avg * (w5_rr - w5_ll)
+        dissipation[3] -= 0.5f0 * λ * h_B_psi * B3_avg * (w5_rr - w5_ll)
+        dissipation[end] -= 0.5f0 * λ * h_B_psi * psi_avg * (w5_rr - w5_ll)
+
+        # Dissipation for the energy equation of species `k` depending on `w_1`, `w_2`, `w_3` and `w_end`. These are the
+        # values of the dissipation that depend on the off-diagonal block ``B`` of the dissipation matrix ``H`` (see equation (80) 
+        # and Figure 1 of the reference given above.
+        ind_E = 3 + (k - 1) * 5 + 5
```

Comment:
```suggestion
        ind_E = 3 + 5 * k
```

Reply:
I suspect that the original notation was intentional to show how this value is computed. Thus maybe rather this:
```suggestion
        ind_E = 3 + 5 * k # simplified version of 3 + (k - 1) * 5 + 5
```